samples/Samples.HttpClientDriver/WebClientHelpers.cs

```diff
@@ -0,0 +1,337 @@
+                    {
+                        webClient.OpenReadAsync(new Uri(url));
+                        while (webClient.IsBusy) ;
+                        Console.WriteLine("Received response for client.OpenReadAsync(Uri)");
+
+                        webClient.OpenReadAsync(new Uri(url), null);
+                        while (webClient.IsBusy) ;
+                        Console.WriteLine("Received response for client.OpenReadAsync(Uri, Object)");
+                    }
+
+                    using (Tracer.Instance.StartActive("OpenReadTaskAsync"))
+                    {
+                        webClient.OpenReadTaskAsync(url).GetAwaiter().GetResult().Close();
+                        Console.WriteLine("Received response for client.OpenReadTaskAsync(String)");
+
+                        webClient.OpenReadTaskAsync(new Uri(url)).GetAwaiter().GetResult().Close();
+                        Console.WriteLine("Received response for client.OpenReadTaskAsync(Uri)");
+                    }
+
+                    using (Tracer.Instance.StartActive("UploadData"))
+                    {
+                        webClient.UploadData(url, new byte[0]);
+                        Console.WriteLine("Received response for client.UploadData(String, Byte[])");
+
+                        webClient.UploadData(new Uri(url), new byte[0]);
+                        Console.WriteLine("Received response for client.UploadData(Uri, Byte[])");
+
+                        webClient.UploadData(url, "POST", new byte[0]);
+                        Console.WriteLine("Received response for client.UploadData(String, String, Byte[])");
+
+                        webClient.UploadData(new Uri(url), "POST", new byte[0]);
+                        Console.WriteLine("Received response for client.UploadData(Uri, String, Byte[])");
+                    }
+
+                    using (Tracer.Instance.StartActive("UploadDataAsync"))
+                    {
+                        webClient.UploadDataAsync(new Uri(url), new byte[0]);
+                        while (webClient.IsBusy) ;
+                        Console.WriteLine("Received response for client.UploadDataAsync(Uri, Byte[])");
+
+                        webClient.UploadDataAsync(new Uri(url), "POST", new byte[0]);
+                        while (webClient.IsBusy) ;
+                        Console.WriteLine("Received response for client.UploadDataAsync(Uri, String, Byte[])");
+
+                        webClient.UploadDataAsync(new Uri(url), "POST", new byte[0], null);
+                        while (webClient.IsBusy) ;
+                        Console.WriteLine("Received response for client.UploadDataAsync(Uri, String, Byte[], Object)");
+                    }
+
+                    using (Tracer.Instance.StartActive("UploadDataTaskAsync"))
+                    {
+                        webClient.UploadDataTaskAsync(url, new byte[0]).GetAwaiter().GetResult();
```

Comment:
Any reason we decided to use `GetAwaiter().GetResult()` instead of `async`/`await` in all of these?

Reply:
I don't know what the reason was, but it was already that way.

Reply:
I'm not sure what the rationale was when I wrote it. I think I also ran into the same initial issue that Bob did and tried to remove async/await keywords to reduce the surface area for troubleshooting purposes.

Reply:
Most users will call these methods using `await`. Are we sure we don't want to test this the same way? (may have to be in a separate PR, but let's create a backlog ticket if so to make sure we're testing this correctly)

Reply:
Good point, yes I think we ultimately should test it that way 👍

Reply:
Added to backlog. Resolving this comment. Thanks everyone for clarifying.